sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -849,6 +849,7 @@ class SessionCatalog(
         tempViews.remove(table)
       }
     }
+    invalidateCachedTable(name)
```

Comment:
Since it was done in the command before, shall we also do it in the v2 command? others may rely on the APIs in the SessionCatalog so we'd better not change it.

Reply:
This is a special logic only for `SesssionCatalog`, should I add special case for it in `DropTableExec`? I think this is a bad start and breaks the generality of V2 command.

Reply:
oh I see, shall we move it to `V2SessionCatalog` then?

Reply:
An obvious flaw now is if people fallback to v1 command, we invalidate the cache twice.

Reply:
> shall we move it to V2SessionCatalog then?

good idea, done.